Update the log forwarding docs (#12567)
newrelic/docs-website: src/content/docs/logs/forward-logs/overview of log-forwarding.mdx

---

> title: Forward your logs to New Relic

Author: ```suggestion
title: Forward host logs to New Relic
```

redirects:
  - /docs/enable-new-relic-logs
  - /docs/logs/new-relic-logs/enable-logs/enable-new-relic-logs
  - /node/36946
  - /docs/logs/enable-logs/enable-logs/enable-new-relic-logs
  - /docs/logs/new-relic-logs/enable-logs/1-enable-new-relic-logs
  - /docs/logs/log-monitoring/enable-logs/enable-log-monitoring-new-relic
  - /docs/logs/enable-log-management-new-relic/enable-log-monitoring-new-relic
  - /docs/logs/enable-log-management-new-relic/enable-log-monitoring-new-relic/enable-log-management-new-relic
  - /docs/logs/new-relic-logs
  - /docs/logs/forward-logs
  - /docs/logs/enable-log-management-in-new-relic/enable-log-monitoring-in-new-relicYES
  - /docs/logs/forward-logs/enable-log-management-new-relicfor
  - /docs/logs/forward-logs/enable-log-management-new-relic
signupBanner:
  text: Monitor and improve your entire stack. 100GB free. Forever.
---

Log forwarding is a way to collect and send your host logs so you can have all your logs in one place. By default, our APM agents collect application-level logs. But if you want to send host logs, you can set up log forwarding depending on what kind of host you have. 

> We offer many options to collect and send host logs, which are listed in the table below. Basically, you can use one of our log fowarding options (i.e., our infrastructure agent, Kubernetes plugin, or log API) or a third-party log forwarder (i.e., AWS, Azure, GCP).

Author: ```suggestion
We offer many options to forward host logs, which are listed in the table below. Basically, you can use one of our log fowarding options (i.e., our infrastructure agent, Kubernetes plugin, or log API) or a third-party log forwarder (i.e., AWS, Azure, GCP).
```

      </td>
    </tr>

    <tr>
      <td>
        Cloud environments
      </td>
      <td>
       * Amazon AWS:
        * [CloudFront plugin](/docs/logs/forward-logs/cloudfront-web-logs)
        * [CloudWatch plugin](/docs/logs/forward-logs/aws-lambda-sending-cloudwatch-logs/)
        * [FireLens plugin](/docs/logs/forward-logs/aws-firelens-plugin-log-forwarding/)
        * [Kinesis Firehose](/docs/logs/forward-logs/stream-logs-using-kinesis-data-firehose/)
        * [AWS Lambda for sending logs from S3](/docs/logs/forward-logs/aws-lambda-sending-logs-s3/)
       * [Google Cloud platform](/docs/logs/forward-logs/google-cloud-platform-log-forwarding)
       * [Microsoft Azure](/docs/logs/forward-logs/azure-log-forwarding/)
       * [Heroku](/docs/logs/forward-logs/heroku-log-forwarding/)
       * [Cloudflare](/docs/logs/forward-logs/cloudflare-logpush-forwarding)
      </td>
    </tr>
    
    <tr>
      <td>
        Docker/Kubernetes containers
      </td>
      <td>
        For Kubernetes environments, use our [Kubernetes plugin](/docs/logs/forward-logs/kubernetes-plugin-log-forwarding/).
        
        For Docker environments, we offer a few options:

>         * Use our [APM logs in context](/docs/apm/new-relic-apm/getting-started/get-started-logs-context)

Author: @lchapman4 @marcelschlapfer I want to move these three bullet points over to the "Forward logs with infra agent" page, as a new section that can be linked to from this table. But for this first bullet point, does this mean APM logs in context automatically collects Docker logs? Or does it mean in conjunction with the infra agent?

        * Use our infrastructure agent. In the `logging.yml` config file be sure to tail the `/var/log/containers/*.log` directory. For example:
          ```yml
          logs:
           - name: Docker-logs
             file: /var/log/containers/*.log # Path to Docker log files
          ```

>         * Use our [Standalone Docker image](https://hub.docker.com/r/newrelic/newrelic-fluentbit-output).

Author: @lchapman4 @marcelschlapfer I want to move these three Docker bullet points over to the "Forward logs with infra agent" page as a new section dedicated to using the infra agent to forward Docker logs. But for this first bullet point, does this mean APM logs in context automatically collects/displays Docker logs? Or does it logs in context data is collected in conjunction with the infra agent?

      </td>
    </tr>
    
    <tr>
      <td>
        Directly from your apps, regardless of where they are hosted
      </td>
      <td>
        * Our <InlinePopover type="apm" /> agent and [infrastructure monitoring agent](/docs/logs/forward-logs/forward-your-logs-using-infrastructure-agent/)
      </td>
    </tr>

    <tr>
      <td>
        Other supported log forwarders
      </td>
      <td>
        * [Our log API](/docs/logs/log-api/introduction-log-api/)
        * Akamai DataStream 2: 
          * [Akamai instructions](https://techdocs.akamai.com/datastream2/docs/stream-new-relic)
          * [Quickstart](https://newrelic.com/instant-observability/datastream2/cb995039-39d8-44d9-97c3-58174a1337f2)
        * Fastly: 
          * [Fastly instructions](https://docs.fastly.com/en/guides/log-streaming-newrelic-logs)
          * [Quickstart](https://newrelic.com/instant-observability/fastly-cdn/c5c5dd30-dcdf-46b6-9412-f9a1bba5a600)
        * [Vector plugin](/docs/logs/forward-logs/vector-output-sink-log-forwarding)

>         * Use the OpenTelemetry SDK to send logs from your apps to an OpenTelemetry collector, which can forward them to New Relic via OTLP.

Author: ```suggestion
        * [Use the OpenTelemetry SDK](add docs link) to send logs from your apps to an OpenTelemetry collector, which can forward them to New Relic via OTLP.
```

>         * Use the OpenTelemetry SDK to send logs from your apps to an OpenTelemetry collector, which can forward them to New Relic via OTLP.

Author: @lchapman4 @marcelschlapfer I'd like to add a docs link with instructions to set this up, but I'm not sure which page for specifically forwarding logs. Can you help me sourcing the best docs link?

Reviewer: Maybe use the link to our infra forwarder? https://docs.newrelic.com/docs/logs/forward-logs/forward-your-logs-using-infrastructure-agent/

Author: But this bullet point is for OpenTelemetry, and we don't have any OTEL info on https://docs.newrelic.com/docs/logs/forward-logs/forward-your-logs-using-infrastructure-agent/ :(
